### PR TITLE
Follow symlinks and defer debug_assertion evaluation

### PR DESCRIPTION
**Pull request by djmarcin**
The large change is refactoring so that rust-embed no longer assumes that it's own compilation flags match those of the expanded code. For example, the proc macro may be compiled in release mode while expanding into code that is compiled in debug mode, or vice versa. This change results in dynamic file lookups when user code is compiled in debug mode, and embedding when compiled in release mode, rather than depending on the compilation of the rust embed proc-macro itself.

The other change included in this PR is to follow symlinks.

Both of these changes are motivated by an attempt to be compatible with the bazel build tool which does not guarantee proc-macro compilation flags match user code compilation flags, and uses symlinks to form a compilation sandbox.

### CODE REVIEW

**Review by pyros2097 — APPROVED** [review]
LGTM as along as it works the same and passes all tests

**Comment by djmarcin** [comment]
> LGTM as along as it works the same and passes all tests

Debug vs non-debug behavior should remain the same, the cfg attributes are just now processed at the time the macro expansion is compiled (e.g. user code) rather than when the proc-macro itself is compiled.

However, following symlinks is a behavior change. I could make the change to put it behind a feature like "follow-symlinks" if you prefer.

File: impl/src/lib.rs [hunk 1]
```diff
@@ -41,7 +96,21 @@ fn generate_assets(ident: &syn::Ident, folder_path: String) -> TokenStream2 {
   }
 }
 
-#[cfg(all(not(feature = "compression"), any(not(debug_assertions), feature = "debug-embed")))]
+fn generate_assets(ident: &syn::Ident, folder_path: String) -> TokenStream2 {
+  let embedded_impl = embedded(ident, folder_path.clone());
+  if cfg!(feature = "rust-embed") {
```
Comment (Mcat12) [inline impl/src/lib.rs:101]: I think this is supposed to be `debug-embed`?

**Review by Mcat12 — CHANGES_REQUESTED** [review]

File: impl/src/lib.rs (hunk 1 above)
Comment (djmarcin) [reply impl/src/lib.rs]: Good catch, thanks. This fails the tests when I run them manually, but it doesn't look like the tests ran automatically on the PR. All pass now.

Comment (Mcat12) [reply impl/src/lib.rs]: @pyros2097 is it expected that fork PRs don't run CI?

Comment (pyros2097) [reply impl/src/lib.rs]: It seems the actions are enabled only on pushes. Will add PR's to it.

Comment (pyros2097) [reply impl/src/lib.rs]: Yeah all the tests are passing

**Review by Mcat12 — APPROVED** [review]

**Pull request merged**